src/Pulsar.Client/Internal/ProducerImpl.fs

```diff
@@ -754,6 +755,42 @@ type internal ProducerImpl<'T> private (producerConfig: ProducerConfiguration, c
             | ProducerMessage.GetStats channel ->
                 channel.SetResult <| stats.GetStats()
 
+            | ProducerMessage.Flush channel ->
+                Log.Logger.LogDebug("{0} Flush requested, pendingMessages count: {1}", prefix, pendingMessages.Count)
+                // First, send all batched messages
+                batchMessageAndSend()
+                
+                // If pendingMessages queue is empty, return immediately
+                if pendingMessages.Count = 0 then
+                    Log.Logger.LogDebug("{0} Flush completed immediately, no pending messages", prefix)
+                    channel.SetResult()
+                else
+                    // Get the last element from the queue by iterating through it
+                    let mutable lastMessage = Unchecked.defaultof<PendingMessage<'T>>
+                    for msg in pendingMessages do
+                        lastMessage <- msg
+                    
+                    Log.Logger.LogDebug("{0} Flush waiting for last message callback, sequenceId: {1}", prefix, %lastMessage.SequenceId)
+                    // Wait for the last message's callback to complete asynchronously
+                    backgroundTask {
+                        match lastMessage.Callback with
+                        | SingleCallback (_, _, tcsOption) ->
+                            match tcsOption with
+                            | Some tcs -> 
+                                let! _ = tcs.Task
```

Comment:
What happens if the task fails? It looks like that FlushAsync will never be completed